.github/workflows/python-app.yml

```
@@ -5,36 +5,45 @@ name: Python application
-        python-version: "3.11"
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install nox
-    - name: Run nox
-      run: |
-        nox
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v3
```

actionlint (1.7.4)

the runner of "actions/setup-python@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)